packages/botonic-plugin-flow-builder/src/api.ts

```diff
@@ -176,4 +177,29 @@ export class FlowBuilderApi {
   private containsAnyKeywords(input: string, keywords: string[]): boolean {
     return keywords.some(keyword => input.includes(keyword))
   }
+
+  getPayload(target?: HtNodeLink): string | undefined {
+    if (target) {
+      if (target.type === HtNodeWithoutContentType.BOT_ACTION) {
+        const botActionNode = this.getNodeById<HtBotActionNode>(target.id)
+        return this.createPayloadWithParams(botActionNode)
+      }
+      const targetNode = this.getNodeById(target.id)
+      return targetNode.id
+    }
+    return undefined
+  }
```

Comment:
As Adrià said I think the best approach would be to implement an abstract class called Target and create two subclasses (BotActionTarget and ContentTarget) which each of this has a method that implements an asbtract method called getPayload. If you want to keep this function I think some improvements can be done:

- Use early returns: `if(!target){ return undefined }`
- You can avoid one getNodeById call: Line 187 is not necessary since you have the target received by param and you can access target.id.

Reply:
I agree with the early return, this way you will remove the long `if(target)` that encapsulates the whole code inside the function.

I also agree with the other point, you could return directly `target.id` and you could avoid the `getNodeById` call.

Regarding the classes, if you do it, do it in a later PR, this one has plenty of stuff already 😉

Reply:
Yeah I don't know why I changed this, I think before I applied this refactor I was debugging the target.id 🤷🏻‍♂️
I will apply the early return and avoid the getNodeById by returning the target.id directly

Reply:
> As Adrià said I think the best approach would be to implement an abstract class with an abstract method getPayload and create two classes (BotActionTarget and ContentTarget) which implement the first one. If you want to keep this function I think some improvements can be done:
> 
> * Use early returns: `if(!target){ return undefined }`
> * You can avoid one getNodeById call: Line 187 is not necessary since you have the target received by param and you can access target.id.

I will keep it in mind for a future refactor to create a Target class that has the logic to create the payloads but right now I won't apply it because it would imply a lot of changes.